crates/pipeline_manager/src/compiler.rs

```diff
@@ -177,18 +178,26 @@ impl Compiler {
 
```

Comment:
is the db.lock() here also starting a transaction on the db?

I put the schema update in the `db.set_project_status_guarded` because I want to ensure the schema exists when status >= CompilingRust to avoid a race but if this is a transaction maybe setting the schema can be a separate query.